src/event_handler.py

```diff
@@ -1,20 +1,41 @@
+import hmac
+import hashlib
+import json
+
 from .github_sdk import GithubClient
 
 
 class EventHandler:
-    def __init__(self, token=None, dsn=None):
-        self.client = GithubClient(token=token, dsn=dsn)
+    def __init__(self, token=None, secret=None, dsn=None, dry_run=False):
+        self.secret = secret
+        self.client = GithubClient(token=token, dsn=dsn, dry_run=dry_run)
 
     def handle_event(self, data, headers):
         # We return 200 to make webhook not turn red since everything got processed well
         http_code = 200
         reason = "OK"
 
-        if headers["X-GitHub-Event"] != "workflow_job":
+        if self.secret and not valid_payload(
+            self.secret, data, headers["X-Hub-Signature"].replace("sha1=", "")
+        ):
+            http_code = 400
```

Comment:
If a developer set ups their Github webhook with a bad secret, the UI for delivered payloads will turn red for each event, thus, making it easy for them to realize that their set up secret is invalid.